src/sagemaker/local/entities.py

```diff
@@ -38,6 +38,149 @@
 HEALTH_CHECK_TIMEOUT_LIMIT = 120
 
 
+class _LocalProcessingJob:
+    """Defines and starts a local processing job."""
+
+    _STARTING = "Starting"
+    _PROCESSING = "Processing"
+    _COMPLETED = "Completed"
+
+    def __init__(self, container):
+        """Creates a local processing job.
+
+        Args:
+            container: the local container object.
+        """
+        self.container = container
+        self.state = "Created"
+        self.start_time = None
+        self.end_time = None
+        self.processing_job_name = ""
+        self.processing_inputs = None
+        self.processing_output_config = None
+        self.environment = None
+
+    def start(self, processing_inputs, processing_output_config, environment, processing_job_name):
+        """Starts a local processing job.
+
+        Args:
+            processing_inputs: The processing input configuration.
+            processing_output_config: The processing input configuration.
+            environment: The collection of environment variables passed to the job.
+            processing_job_name: The processing job name.
+        """
+        self.state = self._STARTING
+
+        for item in processing_inputs:
+            if "DatasetDefinition" in item:
+                raise RuntimeError("DatasetDefinition is not currently supported in Local Mode")
+
+            if "S3Input" in item and item["S3Input"]:
+                data_uri = item["S3Input"]["S3Uri"]
+            else:
+                raise ValueError("Processing input must have a valid ['S3Input']")
```

Comment:
[EAFP](https://docs.python.org/3.8/glossary.html) apply here. suggested edit?

```suggestion
            try:
                data_uri = item["S3Input"]["S3Uri"]
            except KeyError:
                raise ValueError("Processing input must have a valid ['S3Input']")
```

And I only suggested a `try..except` to preserve your intention of a `ValueError` and control of the exception message. Otherwise, remove it and just single line:

```
data_uri = item["S3Input"]["S3Uri"]
```

also apply EAFP here and elsewhere

Reply:
Done, when getting S3Input and S3Output. Other checks should be OK as the keys are optional and we don't want to raise errors.